packages/flutter_tools/lib/src/compile.dart

```diff
@@ -176,8 +176,8 @@ class ResidentCompiler {
 
     final String inputKey = new Uuid().generateV4();
     _server.stdin.writeln('recompile ${mainPath != null ? _mapFilename(mainPath) + " ": ""}$inputKey');
-    for (String filePath in invalidatedFiles) {
-      _server.stdin.writeln(_mapFilename(filePath));
+    for (String fileUri in invalidatedFiles) {
+      _server.stdin.writeln(_mapFilenameToUri(Uri.parse(fileUri).toFilePath()).toString());
```

Comment:
As discussed offline maybe we can avoid the Uri to filepath and back to Uri conversion for the case when _fileSystemRoots is null (more common case) by moving the check for _fileSystemRoots here.

Reply:
Yes, makes sense. Done.